src/hotspot/share/gc/parallel/psCardTable.cpp

```diff
@@ -411,13 +411,8 @@ bool PSCardTable::addr_is_marked_precise(void *addr) {
   return false;
 }
 
-// Assumes that only the base or the end changes.  This allows indentification
-// of the region that is being resized.  The
-// CardTable::resize_covered_region() is used for the normal case
-// where the covered regions are growing or shrinking at the high end.
-// The method resize_covered_region_by_end() is analogous to
-// CardTable::resize_covered_region() but
-// for regions that grow or shrink at the low end.
+// Assumes that only the end changes (the bottom/start stays the same). This
+// allows identification of the region that is being resized.
 void PSCardTable::resize_covered_region(MemRegion new_region) {
   for (int i = 0; i < _cur_covered_regions; i++) {
```

Comment:
This method seems to be completely superfluous now and a call to the parent method achieves exactly the same.

I mean, this code in all cases calls `resize_covered_region_by_start()` in any case, which calls the super class method.

`CardTable::resize_covered_region` performs that `_cur_covered_regions < _max_covered_regions` assert too anyway (in `CardTable::find_covered_region_by_base()`).

So I think this entire method (and `resize_covered_region_by_start`) should be removed too.

I think this is a leftover from removal of the (ever broken, see [here](https://bugs.openjdk.java.net/issues/?jql=text%20~%20%22useadaptivegcboundary%22)) functionality to move the generation boundary (`UseAdaptiveGCBoundary`).

Reply:
Yes, you are right. I have removed those two methods.